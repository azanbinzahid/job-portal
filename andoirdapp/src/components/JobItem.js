import React from 'react';
import {Card, Text, CardItem, Body, Badge} from 'native-base';

const truncate = (str) => {
  return str.length > 200 ? str.substring(0, 180) + '...' : str;
};

const JobItem = (props) => {
  const job = props.job;
  return (
    <Card>
      <CardItem header bordered>
        <Text>{job.title}</Text>
      </CardItem>

      <CardItem bordered>
        <Text note>Company: </Text>
        <Text>{job.company} </Text>
      </CardItem>

      <CardItem bordered>
        <Body>
          <Text note>Description: </Text>
          <Text>{truncate(job.description)}</Text>
        </Body>
      </CardItem>

      <CardItem bordered>
        <Text note>Location: </Text>
        {job.location.map((loc) => (
          <>
            <Badge info>
              <Text>{loc}</Text>
            </Badge>
            <Text> </Text>
          </>
        ))}
      </CardItem>
      <CardItem bordered>
        <Text note>Category: </Text>
        {job.category.map((cat) => (
          <>
            <Badge warning>
              <Text>{cat} </Text>
            </Badge>
            <Text> </Text>
          </>
        ))}
      </CardItem>
      <CardItem
        bordered
        footer
        button
        onPress={() => props.navigation.navigate('Job', {jobId: job.id})}>
        <Text>View Details</Text>
      </CardItem>
    </Card>
  );
};

export default JobItem;
