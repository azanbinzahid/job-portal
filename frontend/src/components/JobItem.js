import React from 'react'
import {  Jumbotron, Button, Card, ListGroup } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';


const JobItem = ({job}) => {

 
    return (
        <Card border="primary">
            <Card.Body>
            <Card.Title>{job.title}</Card.Title>
            <Card.Subtitle className="mb-2 text-muted"> {job.company} </Card.Subtitle>
            <Card.Text>
                {job.description}
            </Card.Text>
            <LinkContainer to={`jobs/${job.id}`}>
                <Button variant="primary"> View Details </Button>
            </LinkContainer>
            </Card.Body>
            <ListGroup variant="flush">
                <ListGroup.Item>{job.location}</ListGroup.Item>
            </ListGroup>
        </Card>
    )
}


export default JobItem;