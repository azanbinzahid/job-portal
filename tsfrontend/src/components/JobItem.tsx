import React from 'react'
import {  Button, Card, ListGroup } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';
import { Job } from 'redux/types';


const truncate =  (str: string) => {
    return str.length > 200 ? str.substring(0, 180) + "..." : str;
}


type Props = {
    job: Job
}


const JobItem = ({job}: Props) => {

 
    return (
        <Card border="primary">
            <Card.Header>
            {
                job.category.map(
                    (cat) => (
                        cat + ", "
                    )
                )
            }
            </Card.Header>
            <Card.Body>
            <Card.Title>{job.title}</Card.Title>
            <Card.Subtitle className="mb-2 text-muted"> {job.company} </Card.Subtitle>
            <Card.Text>
                {truncate(job.description)}
            </Card.Text>
            <LinkContainer to={`jobs/${job.id}`}>
                <Button variant="primary"> View Details </Button>
            </LinkContainer>
            </Card.Body>
            <ListGroup variant="flush">
                {job.location.map( (loc, index) =>(
                    <ListGroup.Item key={index}>{loc}</ListGroup.Item>
                    )
                )}
            </ListGroup>
        </Card>
    )
}


export default JobItem;