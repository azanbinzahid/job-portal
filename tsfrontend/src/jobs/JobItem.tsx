import React, { FC } from "react";
import { Button, Card, ListGroup } from "react-bootstrap";
import { Job } from "redux/types";
import { truncate } from "utils/helper";
import { withRouter, RouteComponentProps } from "react-router-dom";

interface Props extends RouteComponentProps<any> {
  job: Job;
}

const JobItem: FC<Props> = (props) => {
  const { job, history } = props;
  return (
    <Card border="primary">
      <Card.Header>{job.category.map((cat) => cat + ", ")}</Card.Header>
      <Card.Body>
        <Card.Title>{job.title}</Card.Title>
        <Card.Subtitle className="mb-2 text-muted">{job.company}</Card.Subtitle>
        <Card.Text>{truncate(job.description)}</Card.Text>
        <Button
          onClick={() => {
            history.push(`/jobs/${job.id}`);
          }}
          variant="primary"
        >
          View Details
        </Button>
      </Card.Body>
      <ListGroup variant="flush">
        {job.location.map((loc, index) => (
          <ListGroup.Item key={index}>{loc}</ListGroup.Item>
        ))}
      </ListGroup>
    </Card>
  );
};

export default withRouter(JobItem);
