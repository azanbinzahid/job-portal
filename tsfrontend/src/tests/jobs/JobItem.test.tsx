import React from "react";
import { shallow } from "enzyme";
import JobItem from "jobs/JobItem";

const job = {
  id: 1,
  location: ["Lahore"],
  category: ["Customer Service"],
  qualification: ["Bachelors"],
  applicants: ["admin"],
  company: "Modern Tech",
  title: "WordPress Developer",
  salaray: 60000,
  description: "This is test job description",
  experiance: 1,
  datestamp: new Date(11 / 11 / 11),
};

test("renders the component", () => {
  const component = shallow(<JobItem job={job} />);
  expect(component).toMatchSnapshot();
});
