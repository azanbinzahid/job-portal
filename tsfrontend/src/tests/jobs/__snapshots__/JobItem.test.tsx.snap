// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Card
  body={false}
  border="primary"
>
  <CardHeader>
    Customer Service, 
  </CardHeader>
  <CardBody>
    <CardTitle>
      WordPress Developer
    </CardTitle>
    <CardSubtitle
      className="mb-2 text-muted"
    >
      Modern Tech
    </CardSubtitle>
    <CardText>
      This is test job description
    </CardText>
    <withRouter(LinkContainer)
      to="jobs/1"
    >
      <Button
        active={false}
        disabled={false}
        variant="primary"
      >
         View Details 
      </Button>
    </withRouter(LinkContainer)>
  </CardBody>
  <ListGroup
    variant="flush"
  >
    <ListGroupItem
      active={false}
      disabled={false}
      key="0"
    >
      Lahore
    </ListGroupItem>
  </ListGroup>
</Card>
`;
